common/src/Common/Wallet.hs

```diff
@@ -241,6 +242,8 @@ data AccountStatus a
 data AccountGuard
   = AccountGuard_KeySet (Set PublicKey) Text
   -- ^ Keyset guards
+  | AccountGuard_KeySetRef Text
```

Comment:
This new constructor seems problematic because it creates incomplete pattern matches in other parts of the code.  Maybe now is a good time to think about the comment above about replacing this with pact's representation.

Although...maybe this is a good opportunity for a middle ground.  Perhaps this AccountGuard data type can become the abstraction we're needing now that abstracts over the difference between KeySet and KeySetRef.

```
data KeySetHeritage = KeySetHeritage
  { keySetHeritage_keys :: Set PublicKey
  , keySetHeritage_pred :: Text
  , keySetHeritage_ref :: Maybe Text
  }

data AccountGuard
  = AccountGuard_KeySetLike KeySetHeritage
  | AccountGuard_Other (Pact.Guard PactValue)
```

This abstraction would make it so that after you look up a keyset-ref's underlying keyset, any code that is downstream from that point will behave the same.  That's probably what we want here, otherwise we'll probably encounter this same bug again in the future.

Reply:
I agree. I've included this data definition in the code.